### PR TITLE
#195 (Infix to Postfix expression conversion(application of stack))

### PR DESCRIPTION
**Pull request by VaishnaviMandloi**
## Related Issue
   -Infix to Postfix expression conversion(application of stack) #195 

## Proposed Changes
- Change 1
    added implementation code for Infix to Postfix expression conversion with comments (explanation) and sample expression.

- Change 2
   added a explanation for Infix to Postfix expression conversion in README file with links and image.

## Additional Info
- It is a application of stack
-  Its Time Complexity  is O(n) , where n is length of expression .
- changes are made only into S-Stack folder.
- tested with many input , screen shot of one of those is provided below.

- algorithm that I used :
1.  Using cpp built stack , For stack operation . 
2. check each character of Infix expression one by one from left to right.
3. If the character is an operand, add to output.
    Else If the character is ‘(‘, pushed in stack.
    Else If the character is ‘)’, pop the stack and output the string upto character '(' .
    Else, it is a operator , check precedence and push or pop accordingly
         If the precedence of the checked operator is greater than the precedence of the operator in the stack(or the stack is 
              empty or the stack contains a ‘(‘ ), push it.
         Else, Pop all the operators from the stack which are greater than or equal to in precedence than that of the checked 
               operator. 
         After doing that Push the checked operator to the stack. (If there is a parenthesis while popping then stop there and push 
         the checked operator in the stack.)
4. Repeat steps 3-6 until the end of Infix expression.
5. pop the remaining expression from the stack until stack is empty
6. print the result string that is the postfix notation.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| code screenshot | output screenshot |
| code #195 learncpp  |    output #195 learncpp |

### CODE REVIEW

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
There are a lot of blank spaces in the cpp file. Make sure you delete these spaces. Also try to follow the readme structure and explain more about the algo !

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
Please avoid copy pasting the information from gfg. Try to write it on your own !

**Comment by siddhi-244** [comment]
///--------------------Pull Requests--------------------///

1. Short Description or CheckLists

Make sure your Pull Request is descriptive and complete the checklist .
I would recommend you to go through this tutorial video of the project [here](https://youtu.be/kIFwqp9B7pQ) ,thanks.


///------------------Description -----------------------///

For contribution and more guidelines i would recommend you to go through this 
tutorial video of the project [here](https://youtu.be/7UgC3HDETRA), thanks.


///-----------------CheckList-------------------------///


Make sure your Pull Request is descriptive and complete the checklist .
I would recommend you to go through this tutorial video of the project [here](https://youtu.be/kIFwqp9B7pQ) ,thanks.

**Comment by VaishnaviMandloi** [comment]
Now I have made changes as mentioned by you please check .

File: S-Stack/InfixToPostfix.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            a^[b*c-d/[e+f]]
+            a^[b*c-d/ef+]
+            a^[bc*-d/ef+]
+            a^[bc*-def+/]
+            a^[bc*def+/-]
+            abc*def+/-^
+*/
+// Main Function
+int main(){
+    string expression ="a^(b*c-d/(e+f))";
```
Comment (siddhi-244) [inline S-Stack/InfixToPostfix.cpp:99]: Can uh take the string as input from the user here?

Comment (VaishnaviMandloi) [reply S-Stack/InfixToPostfix.cpp:99]: Okay I will modify the code

**Comment by VaishnaviMandloi** [comment]
@siddhi-244 Now is it better ? please check .

**Review by siddhi-244 — APPROVED** [review]
LGTM

**Comment by VaishnaviMandloi** [comment]
thanks @siddhi-244

**Comment by VaishnaviMandloi** [comment]
@siddhi-244 I updated the description if it is enough please remove the tag of short description

**Comment by VaishnaviMandloi** [comment]
@iamgunjan please review it.

**Comment by Lakhankumawat** [comment]
1. Star the repo 
2. Uncheck all the points in checklists which do not apply to you

**Comment by VaishnaviMandloi** [comment]
@Lakhankumawat  Is it now correct?

**Comment by Lakhankumawat** [comment]
LGTM 💯

**Comment by Lakhankumawat** [comment]
@all-contributors please add @VaishnaviMandloi  code ,

**Pull request merged**